storage/connect/jdbconn.cpp

```diff
@@ -451,8 +451,9 @@ PQRYRES JDBCSrcCols(PGLOBAL g, PCSZ src, PJPARM sjp)
 
 	if (strstr(src, "%s")) {
 		// Place holder for an eventual where clause
-		sqry = (char*)PlugSubAlloc(g, NULL, strlen(src) + 2);
-		sprintf(sqry, src, "1=1");			 // dummy where clause
+		size_t sqry_size = strlen(src) + 2;
+		sqry = (char*)PlugSubAlloc(g, NULL, sqry_size);
+		snprintf(sqry, ((sqry_size + 7) / 8) * 8, src, "1=1");			 // dummy where clause
```

Comment:
what is the additional maths here?

Reply:
PlugSubAlloc(,,buf_size) function allocate memory  size = buf_size but round up the size to multiple of 8.

Reply:
> There is however the case where modern compilers understand printf enough to warn about sizes.
Yes, that's true but I consider the code can be changed here and we can have not a static but dynamic string. In any case it's better not to have unsafe function in the code if possible

Reply:
> what is the additional maths here?

PlugSubAlloc(..buf_size) allocate buffer of size = buf size but round up it to multiply of 8

Reply:
Further to what @grooverdan said here, if you could make this a macro with a semi-descriptive name then it would be easier to understand to the casual observer.

Reply:
> Further to what @grooverdan said here, if you could make this a macro with a semi-descriptive name then it would be easier to understand to the casual observer.

Added const int instead and added comment explaining the code